spec/mongoid/traversable_spec.rb

```diff
@@ -251,4 +251,164 @@
+
+      it "is changed in the child: Piano" do
+        expect(Piano.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the child: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello")
+      end
+    end
+
+    context "when the discriminator key is changed in the parent" do
+      before do
+        Instrument.discriminator_key = "hello2"
+      end
+
+      after do
+        Instrument.discriminator_key = nil
+      end
+
+      it "the global discriminator key is _type" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "the value changes" do
+        expect(Instrument.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Piano" do
+        expect(Piano.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello2")
+      end
+
+      context 'when discriminator key is set to nil in parent' do
+        before do
+          Instrument.discriminator_key = nil
+        end
+
+        it "the global discriminator key is _type" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'uses global setting' do
+          expect(Instrument.discriminator_key).to eq("_type")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("_type")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("_type")
+        end
+      end
+
+      context "when resetting the discriminator key after nil" do
+        before do
+          Instrument.discriminator_key = "hello4"
+        end
+
+        it "the global discriminator key is _type" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'has the correct value' do
+          expect(Instrument.discriminator_key).to eq("hello4")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("hello4")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("hello4")
+        end
+      end
+    end
+
+    context "when the discriminator key is changed in the child" do
+      it "raises an error" do
+        expect do
+          Guitar.discriminator_key = "hello3"
+        end.to raise_error(Mongoid::Errors::InvalidDiscriminatorKeyTarget)
+      end
+
+      it "doesn't change in that class" do
+        expect(Guitar.discriminator_key).to eq("_type")
+      end
+
+      it "the global discriminator key is _type" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "doesn't change in the sibling" do
+        expect(Piano.discriminator_key).to eq("_type")
+      end
+
+      it "doesn't change in the parent" do
+        expect(Instrument.discriminator_key).to eq("_type")
+      end
+    end
```

Comment:
I think you need to add some sort of `before` block that tries to set the discriminator key on `Guitar`.

Reply:
I changed it to a let block... does that work?